File: micrometer-core/src/test/groovy/io/micronaut/docs/MetricOptionsFilterTaggersExample.java
package io.micronaut.docs;

import io.micrometer.core.annotation.Timed;
import io.micronaut.configuration.metrics.annotation.MetricOptions;
import jakarta.inject.Singleton;

@Singleton
public class MetricOptionsFilterTaggersExample {

    @MetricOptions(
        filterTaggers = true, // Specify that not all taggers should be applied
        taggers = {MethodNameTagger.class} // Specific taggers to apply
    )
    @Timed(value = "do_something")
    public void doSomething() {
        // ...
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/metrics/annotation/MetricOptionsFilterTaggersExample.kt
package io.micronaut.docs.metrics.annotation

// tag::imports[]
import io.micrometer.core.annotation.Timed
import io.micronaut.configuration.metrics.annotation.MetricOptions
import jakarta.inject.Singleton
// end::imports[]

// tag::class[]
@Singleton
open class MetricOptionsFilterTaggersExample {

    @MetricOptions(
        filterTaggers = true, // Specify that not all taggers should be applied
        taggers = [MethodNameTagger::class] // Specific taggers to apply
    )
    @Timed(value = "do_something")
    open fun doSomething() {
        // ...
    }
}
// end::class[]
